src/Microsoft.PowerApps.TestEngine/PowerFx/PowerFxHelper.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerApps.TestEngine.PowerFx
+{
+    /// <summary>
+    /// Static entity with methods to help with any powerfx related operations
+    /// </summary>
+    public static class PowerFxHelper
+    {
+        /// <summary>
+        /// Extracts the formulas separated by chaining/variadic operator.
+        /// Operator is ";" when decimal separator for the locale is "." and is ";;" when decimal separator is ","
+        /// </summary>
+        /// <param name="engine">Instance of an engine configured with the desired locale</param>
+        /// <param name="expression">Expression from which formulas separated by chaining operator would be extracted</param>
+        /// <returns>An enumerable of formulas extracted from the expression that are separated by chaining operator</returns>
+        public static IEnumerable<string> ExtractFormulasSeparatedByChainingOperator(Engine engine, string expression)
+        {
+            if (string.IsNullOrWhiteSpace(expression))
+            {
+                return new string[0];
+            }
+
+            var chainOperatorTokens = engine.Tokenize(expression).Where(tok => tok.Kind == TokKind.Semicolon).OrderBy(tok => tok.Span.Min);
```

Comment:
what is the significance of orderby span.min ? what value is that ?

Reply:
Span.Min is the start of the span. This ensures that we order the semicolon tokens by their spans so we process from left to right. Tokens might be sorted already but this makes up future proof. Ideally this logic is powerfx specific and doesn't have anything test engine specific so it should live inside powerfx but that is out of scope for this. if we persist with this hack and need this logic here then we could consider moving it inside the powerfx